### PR TITLE
Google Cloud Run is not for Background jobs

### PR DESCRIPTION
**Pull request by evantahler**

### CODE REVIEW

File: pages/blog/google-cloud-run-no-background-job.mdx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+title: "Google Cloud Run does not support Background Jobs"
+date: "2021-04-13"
+author: "Evan Tahler"
+pullQuote: "Google Cloud Run is a great platform as a service, but not for background jobs"
+tags: [engineering, notes]
+---
+
+Grouparoo is a self-hosted product, so we are always looking for the simplest ways to help our customers run the application. A new member of the Google Cloud Platform (GCP) family is [Google Cloud Run](https://cloud.google.com/run) - which is the closest Google has come yet to a Heroku-like "Git-Ops" way to deploy your applications. It handles load balancing, scaling, and more for you and is really compelling product. Combined with [Google Cloud Build](https://cloud.google.com/build), you can wire up your service to (re)deploy automatically when your git repository changes.
```
Comment (andyjih) [inline pages/blog/google-cloud-run-no-background-job.mdx]:
```suggestion
Grouparoo is a self-hosted product, so we are always looking for the simplest ways to help our customers run the application. A new member of the Google Cloud Platform (GCP) family is [Google Cloud Run](https://cloud.google.com/run) - which is the closest Google has come yet to a Heroku-like "Git-Ops" way to deploy your applications. It handles load balancing, scaling, and more for you and is a really compelling product. Combined with [Google Cloud Build](https://cloud.google.com/build), you can wire up your service to (re)deploy automatically when your git repository changes.
```

File: pages/blog/google-cloud-run-no-background-job.mdx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+After those steps, and setting our environment variables, we had Grouparoo running on Google Cloud Run!
+
+But... it didn't last long. Every few hours, we would notice that our job throughput would grind to 0. We would then visit the site to look for failures, but everything appeared to be OK, and would start working again. However, only a few hours later, things would slow down again. What was going on?
+
+After some digging, we learned that Google Cloud Run throttled based on HTTP requests, and _only_ HTTP requests.
+
+> When an application running on Cloud Run finishes handling a request, the container instance's access to CPU will be disabled or severely limited. Therefore, you should not start background threads or routines that run outside the scope of the request handlers.
+
+This makes Google Cloud Run a poor platform choice for an application like grouparoo which manages its own background jobs, and expects at least one instance to always be available for scheduling. You can learn more [here](https://cloud.google.com/run/docs/tips/general#avoiding_background_activities). That explains why when we visited the site, things started working again, and stopped after that.
```
Comment (andyjih) [inline pages/blog/google-cloud-run-no-background-job.mdx]:
```suggestion
This makes Google Cloud Run a poor platform choice for an application like Grouparoo which manages its own background jobs, and expects at least one instance to always be available for scheduling. You can learn more [here](https://cloud.google.com/run/docs/tips/general#avoiding_background_activities). That explains why when we visited the site, things started working again, and stopped after that.
```

**Review by andyjih — APPROVED** [review]
Looks good! Just a few minor tweaks.

**Comment by evantahler** [comment]
Thanks @andyjih !

**Pull request merged**